### ISSUE TITLE
Adding more functionality to our `BigInt` Library

### ISSUE THREAD

**Issue by davekaj** [graphprotocol/graph-ts#35]
A list of improvements that would be nice to have:

- [ ] Comparisons -  implement `greaterThan(),` `lessThan()` `equalTo`, `greaterThanOrEqualTo()` , etc.
- [ ] absolute value function - `.abs()`
- [ ] the length of the string, so `123456789123456789` would return 18 (we might have this already actually, please let me know)
- [ ]  an `isI32()` func would be nice. So if it fits in an `i32` we can convert it down to i32, but if it is too too big then we return false, and can avoid crashing
- [ ] an `isZero()` method
- [x] ~~`BigInt.toString` actually returns the string value, and does not convert to i32 first (see explanation below)~~

**BigInt.toString() notes** 
BigInt.toString() is misleading since it first converts to i32, then to string
You would think you could get the pure string of the BigInt, but you can't. If you do:

`BigInt.fromI32(4149483648).toString()` 

It will return `"-145483648"`.

I think it would be better to actually turn it directly into a string. 

In the mean time, when we make docs for `graph-ts` this quirk should be mentioned.

**Comment by davekaj** [comment]
fyi I found this in the DAOstack repo, it is a funciton to compare big ints:
```
export function equals(a: BigInt, b: BigInt): boolean {
  if (a.length !== b.length) {
    return false;
  }
  for (let i = 0; i < a.length; i++) {
    if (a[i] !== b[i]) {
      return false;
    }
  }
  return true;
}
```

**Comment by Jannis** [comment]
@davekaj `BigInt.fromI32(4149483648).toString()` returning `"-145483648"` is correct. `4,149,483,648` is larger than the maximum value that an i32 can represent (`2,147,483,647`) and so it'll overflow.

You can play with it here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2b9507443bb8c0e2572033655e9b5ec4 — note that if you disable the `#[allow(overflowing_literals)]`, the compiler will fail because the value is out of range for `i32`.

**Comment by davekaj** [comment]
Ah I see my Mistake. What I was intending to say was `BigInt.toString()` under the hood, was first converting to an I32, and then to a string. So I was trying to say that this was misleading. 

But, I think I must have done `BigInt.fromI32()` and just _assumed_ that it returned me the correct `BigInt`, and that `toString()` was then converting it to I32.

BUT obviously, it was the fact that I was trying to create a BigInt with `fromI32()`, which from the get go gave me the negative number, and `toString()` correctly turned that into a negative string. 

To conclude: both `fromI32()` and `toString()` work properly. When I combined them I got confused and thought `toString()` was broken.
